chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/DeveloperStatusEventsHelper.java

```diff
@@ -0,0 +1,117 @@
+            final List<DeveloperStatusEventDTO> updatedEvents) {
+        List<DeveloperStatusEventDTO> originalStatuses =
+                cloneDeveloperStatusEventList(originalEvents);
+        List<DeveloperStatusEventDTO> updatedStatuses =
+                cloneDeveloperStatusEventList(updatedEvents);
+
+        updatedStatuses.removeAll(originalStatuses);
+        return updatedStatuses;
+    }
+
+    /**
+     * Determine if an event was removed.
+     * @param originalEvents - List<DeveloperStatusEventDTO>
+     * @param updatedEvents - List<DeveloperStatusEventDTO>
+     * @return List<DeveloperStatusEventDTO>
+     */
+    public static List<DeveloperStatusEventDTO> getRemovedEvents(final List<DeveloperStatusEventDTO> originalEvents,
+            final List<DeveloperStatusEventDTO> updatedEvents) {
+        List<DeveloperStatusEventDTO> originalStatuses =
+                cloneDeveloperStatusEventList(originalEvents);
+        List<DeveloperStatusEventDTO> updatedStatuses =
+                cloneDeveloperStatusEventList(updatedEvents);
+
+        originalStatuses.removeAll(updatedStatuses);
+        return originalStatuses;
+    }
+
+    /**
+     * Determine if an event was updated.
+     * @param originalEvents - List<DeveloperStatusEventDTO>
+     * @param updatedEvents - List<DeveloperStatusEventDTO>
+     * @return List<DeveloperStatusEventDTO>
+     */
+    public static List<DeveloperStatusEventPair> getUpdatedEvents(final List<DeveloperStatusEventDTO> originalEvents,
+            final List<DeveloperStatusEventDTO> updatedEvents) {
+        List<DeveloperStatusEventPair> statuses = new ArrayList<DeveloperStatusEventPair>();
+        for (DeveloperStatusEventDTO updatedStatusEvent : updatedEvents) {
+            for (DeveloperStatusEventDTO existingStatusEvent : originalEvents) {
+                if (updatedStatusEvent.equals(existingStatusEvent)) {
+                    if (!nullSafeEquals(updatedStatusEvent.getReason(), existingStatusEvent.getReason())
+                            || !nullSafeDateEquals(
+                                    updatedStatusEvent.getStatusDate(), existingStatusEvent.getStatusDate())
+                            || !nullSafeEquals(updatedStatusEvent.getStatus().getStatusName(),
+                                    existingStatusEvent.getStatus().getStatusName())) {
+
+                        statuses.add(new DeveloperStatusEventPair(existingStatusEvent, updatedStatusEvent));
+                    }
+                }
+            }
+        }
+
+        return statuses;
+    }
+
+    private static boolean nullSafeDateEquals(final Date obj1, final Date obj2) {
+        if (obj1 == null || obj2 == null) {
+            return true;
+        } else {
+            Calendar cal1 = Calendar.getInstance();
+            cal1.setTime(obj1);
+            Calendar cal2 = Calendar.getInstance();
+            cal2.setTime(obj2);
+            if (cal1.get(Calendar.MONTH) == cal1.get(Calendar.MONTH)
```

Comment:
Do you mean to compare cal1 to cal2 here?

Reply:
Also not sure if it will work with Date objects but I often use apache's ObjectUtil.equals method which is null safe - might be worth a test to see if it handles Dates.

Reply:
Yes - it should be cal1 compared to cal1.

I only want to compare the year, month, and day portion of the date - excluding the time portion, so I don't think ObjectUtils.equals will work.

Reply:
I'm confused, cal1 month will always equal cal1 month (day and year too). Shouldn't it compare cal1 month to cal2 month (day and year)?

Reply:
Ugh - forgot to make the code change...